build.sh

```diff
@@ -86,4 +84,4 @@ export CI_SNAPSHOT_RELEASE="$projectPrefix/publish"
 # for now, until we're confident in the new release scripts, just close the staging repo.
 export CI_SONATYPE_RELEASE="; sonatypePrepare; sonatypeBundleUpload; sonatypeClose"
 
-sbt "$setTagScalaVersion" clean $projectPrefix/test $projectPrefix/publishLocal $releaseTask
```

Comment:
I think this is wrong (in the current setup). If I create a tag `v1.2.3#2.13.0-M3`, this should override the scala version defined in `TRAVIS_SCALA_VERSION`, that's the core of the feature.

To simplify the scripts, I'd be fine dropping the `tagScalaVer` (the ability to re-release just by adding the scala version to the tag, like `v1.2.3#2.13.0-M3`) and everything related completely. This would mean that, if you want to re-release an existing scala-xml for a new Scala version (or a new Scala.js version), you take the commit of the corresponding scala-xml tag, change the `.travis.yml` to only include the new Scala (Scala.js) version(s) that you want to re-build for, and create a new tag for this commit.

Mabye this is the best way forward, given that binary incompatible Scala or Scala.js releases are not going to be frequent in the future. cc @SethTisue, what do you think?

Reply:
So the assumption with the feature, the script setting `scalaVersion` explicitly, is that you would just create the tag and you would publish artifacts without having to commit anything or change the build at all?  Does that happen often? That has not been the case in my experience, but maybe I've been doing things wrong.

Reply:
> you would just create the tag and you would publish artifacts without having to commit anything or change the build at all?

Yes, that is/was the idea. It worked for a few times, especially during milestone/RC cycles, where each release is binary incompatible, and they're relatively close to each other. But now that this situation is pretty far out, it's probably simpler to just remove the feature.

Reply:
It seems worth keeping around for publishing Dotty RC artifacts.

Reply:
Now that we're on sbt-ci-release, each target gets its own staging repo. (Before, we used to get an unpredictable number containing an unpredictable mix of files.) So one way to handle this is to just publish everything, then drop the unwanted staging repos. Throwing the idea out there.

Reply:
So use `ci-release` instead of `publishSigned?  I'm happy to add that here and give it a try.

Reply:
We're doing that already https://github.com/scala/scala-xml/blob/c323a319056d2fcb55c9a7524aaa142ca36002e8/build.sh#L58

Reply:
Oh ok, so what is @SethTisue suggesting?

Reply:
One issue with `v1.2.3#2.13.0-M3` is that in `.travis.yml` we have

```
scala:
  - 0.22.0-RC1
  - 2.12.10
  - 2.13.1
```

We override the Scala version with the one in the tag, but we're still running separate travis builds for each version listed in the config. So instead of aborting some of these builds (what the script currently does), we could just drop the staging repos on sonatype.

I'm personally convinced by now to drop all of this complexity and just require a new commit that adjusts `.travis.yml`, seems to be the easiest solution.

Reply:
Oh, so he's suggesting running `sonatypeDrop` instead of aborting the script with `exit 0`?

Reply:
@ashawley I'm suggesting: leave the script alone, let all combinations publish, and then in the Sonatype web UI, manually choose which repos to drop or release.

Lukas's suggestion (of pushing a commit that omits the combinations we don't want) works fine too.

Reply:
Ok, sounds like you and Lukas are suggesting the same thing.   I rely on Lightbend team for releases, so I defer to you.  I'll work on ripping it out.  I'll just need to keep the Java version checks.

Reply:
fwiw, I opened https://github.com/olafurpg/sbt-ci-release/issues/102 to see if Olaf has an “official” position on this

Reply:
That's a good idea!  Thanks, Seth.